File: packages/react-native-autoplay/ios/hybrid/HybridSearchTemplate.swift
//
//  HybridSearchTemplate.swift
//  Pods
//
//  Created by Samuel Brucksch on 28.10.25.
//

import NitroModules

class HybridSearchTemplate: HybridSearchTemplateSpec {
    func createSearchTemplate(config: SearchTemplateConfig) throws {
        let template = SearchTemplate(config: config)
        TemplateStore.addTemplate(
            template: template,
            templateId: config.id
        )
    }

    func updateSearchResults(templateId: String, results: NitroSection) throws
        -> Promise<Void>
    {
        return Promise.async {
            guard
                let template = TemplateStore.getTemplate(templateId: templateId)
                    as? SearchTemplate
            else {
                throw AutoPlayError.invalidTemplateType(
                    "\(templateId) is not a SearchTemplate"
                )
            }

            await template.updateSearchResults(results: results)
        }
    }
}

File: packages/react-native-autoplay/android/src/main/java/com/margelo/nitro/swe/iternio/reactnativeautoplay/HybridSearchTemplate.kt
package com.margelo.nitro.swe.iternio.reactnativeautoplay

import com.margelo.nitro.core.Promise
import com.margelo.nitro.swe.iternio.reactnativeautoplay.template.AndroidAutoTemplate
import com.margelo.nitro.swe.iternio.reactnativeautoplay.template.SearchTemplate

class HybridSearchTemplate : HybridSearchTemplateSpec() {

    override fun createSearchTemplate(config: SearchTemplateConfig) {
        val context = AndroidAutoSession.getRootContext()
            ?: throw IllegalArgumentException("createSearchTemplate failed, carContext not found")

        val template = SearchTemplate(context, config)
        AndroidAutoTemplate.setTemplate(config.id, template)
    }

    override fun updateSearchResults(templateId: String, results: NitroSection): Promise<Unit> {
        return Promise.async {
            val template = AndroidAutoTemplate.getTemplate<SearchTemplate>(templateId)
            template.updateSearchResults(results)
        }
    }
}